vespalib/src/vespa/vespalib/net/tls/protocol_snooping.cpp

```diff
@@ -0,0 +1,128 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#include "protocol_snooping.h"
+#include <iostream>
+#include <cstdlib>
+#include <stdint.h>
+
+namespace vespalib::net::tls {
+
+namespace {
+
+// Precondition for all helper functions: buffer is at least `min_header_bytes_to_observe()` bytes long
+
+// From RFC 5246:
+// 0x16 - Handshake content type byte of TLSCiphertext record
+// 0x03 - First byte of 2-byte ProtocolVersion, always 3 on TLSv1.2 and v1.3
+inline bool is_tls_handshake_packet(const char* buf) {
+    return ((buf[0] == 0x16) && (buf[1] == 0x03));
```

Comment:
I thought the second byte was the major version and should go into the version check and not the handshake tag check?

Reply:
Yes, have moved it to the right place. Was part of the original "2 initial bytes check" approach so it just carried over :)